### ISSUE TITLE
Are contents after ``\end{document}'' ignored?

### ISSUE THREAD

**Issue by lirm-math** [SublimeText/LaTeXTools#1234]
In LaTeX, all the contents after `\end{document}` are regarded as comments. 
But it seems that in the current version (ST 3, LaTeXTools v3.14.2) they are not ignored. 

When I type `\cite{}` or `C-R`, it looks for information from after `\end{document}`.

**Comment by r-stein** [comment]
Yes that's a design choice we changed a few updates ago. The reason for this is that some people are using `\end{document}` inside verbatim or listing environments/commands and we can't fix that for every case. So this is IMO the better option.

**Comment by Romeoly** [comment]
Yes, it is hard to tell what are "fake" `\end{document}`s. So, can we just try to figure out what is a "real" one? For example, can we tell `LaTeXTools` that this is really the ``\end{document}" by comments like 

`
\end{document}  % THIS_IS_THE_END_OF_THE_DOCUMENT
` 

Or maybe we can set in `Settings - User` what is a "real" `\end{document}`?

**Comment by r-stein** [comment]
I thought about adding comment hints for the LaTeXTools analysis, such that you can just add citations, references, etc. the analysis does not understand via comments. These could also stop the analysis.

**Comment by Romeoly** [comment]
If the analysis is stopped by the % char, that won't be a problem. 

The hint does not have to be started with a %, which might confuse the analyser. 

`\end{document} THIS_IS_THE_END_OF_THE_DOCUMENT`

is just as good, because LaTeX neglects everying after `\end{document}` even in the same line.
